parsons/azure/azure_blob_storage.py

```diff
@@ -0,0 +1,390 @@
+import logging
+import os
+from urllib.parse import urlparse
+
+from azure.core.exceptions import ResourceNotFoundError
+from azure.storage.blob import BlobServiceClient, ContentSettings, generate_blob_sas
+
+from parsons.utilities import check_env, files
+
+logger = logging.getLogger(__name__)
+
+
+class AzureBlobStorage(object):
+    """
+    Instantiate AzureBlobStorage Class for a given Azure storage account.
+
+    `Args:`
+        account_name: str
+            The name of the Azure storage account to use. Not required if ``AZURE_ACCOUNT_NAME``
+            environment variable is set, or if ``account_url`` is supplied.
+        credential: str
+            An account shared access key with access to the Azure storage account, an SAS token
+            string, or an instance of a TokenCredentials class. Not required if ``AZURE_CREDENTIAL``
+            environment variable is set.
+        account_domain: str
+            The domain of the Azure storage account, defaults to "blob.core.windows.net".
+            Not required if ``AZURE_ACCOUNT_DOMAIN`` environment variable is set or if
+            ``account_url`` is supplied.
+        account_url: str
+            The account URL for the Azure storage account including the account name and domain.
+            Not required if ``AZURE_ACCOUNT_URL`` environment variable is set.
+    `Returns:`
+        `AzureBlobStorage`
+    """
+
+    def __init__(self, account_name=None, credential=None, account_domain='blob.core.windows.net',
+                 account_url=None):
+        self.account_name = check_env.check('AZURE_ACCOUNT_NAME', account_name)
+        self.credential = check_env.check('AZURE_CREDENTIAL', credential)
+        self.account_domain = check_env.check('AZURE_ACCOUNT_DOMAIN', account_domain)
+        self.account_url = os.getenv('AZURE_ACCOUNT_URL', account_url)
+        if not self.account_url:
+            self.account_url = f'https://{self.account_name}.{self.account_domain}/'
+        else:
+            if not self.account_url.startswith('http'):
+                self.account_url = f'https://{self.account_url}'
+            # Update the account name and domain if a URL is supplied
+            parsed_url = urlparse(self.account_url)
+            self.account_name = parsed_url.netloc.split(".")[0]
```

Comment:
I think if you move this block up before the `check_env` call for `account_name` and `account_domain`, then a user could pass in an `account_url` and not a `account_name` without the `check_env` raising an `Exception`.

Reply:
Ah good point, thanks for catching that! I'll make that change